apps/app-backend/src/functions/get-service-by-id.ts

```diff
@@ -0,0 +1,95 @@
+import * as H from "@pagopa/handler-kit";
+import * as L from "@pagopa/logger";
+import * as E from "fp-ts/lib/Either";
+import * as O from "fp-ts/lib/Option";
+import * as RTE from "fp-ts/lib/ReaderTaskEither";
+import * as TE from "fp-ts/lib/TaskEither";
+
+import { flow, pipe } from "fp-ts/lib/function";
+
+import { httpAzureFunction } from "@pagopa/handler-kit-azure-func";
+import { readableReport } from "@pagopa/ts-commons/lib/reporters";
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import { ServiceDetails } from "../generated/definitions/internal/ServiceDetails";
+import { PathParamValidatorMiddleware } from "../middleware/path-params-middleware";
+import { ServiceDetailsContainerDependency } from "../utils/cosmos-db/dependency";
+
+/**
+ * GET /services/{serviceId} AZF HttpTrigger
+ * Get Service by Id
+ */
+
+const executeGetServiceById: (
+  serviceId: NonEmptyString
+) => RTE.ReaderTaskEither<
+  ServiceDetailsContainerDependency,
+  H.HttpError,
+  ServiceDetails
+> =
+  (serviceId: NonEmptyString) =>
+  ({ serviceDetailsContainer }) =>
+    pipe(
+      TE.tryCatch(
+        () => serviceDetailsContainer.item(serviceId, serviceId).read(),
+        (err) =>
+          new H.HttpError(
+            `An error has occurred while fetching service having ID ${serviceId} [${
+              E.toError(err).message
+            }]`
+          )
+      ),
+      TE.chain((rr) =>
+        pipe(
+          rr,
+          O.fromPredicate(() => rr.statusCode === 200),
+          O.fold(
+            () =>
+              E.left(
+                new H.HttpNotFoundError(`Service '${serviceId}' not found`)
```

Comment:
**question (blocking)**: are you sure this case happens only for not found item?

I mean, the response status code can be different from either 200 or 404 (rarely, but it can happen).

Reply:
598a7c503c5391f309586d5b3cdc5cf51cd243ab and 808189fd3f654572d64a287b3da1970fe7f21952